src/index.ts

```diff
@@ -877,7 +888,9 @@ export class BufferedChangeset implements IChangeset {
       // 'user.name'
       const normalizedBaseChanges = normalizeObject(baseChanges);
       if (isObject(normalizedBaseChanges)) {
-        const result = this.getDeep(normalizedBaseChanges, remaining.join('.'));
+        const result = this.maybeUnwrapProxy(
+          this.getDeep(normalizedBaseChanges, remaining.join('.'))
+        );
```

Comment:
This will prevent us from wrapping an ObjectProxy with a value of null in an `ObjectTreeNode`.